packages/react/use-body-pointer-events/src/useBodyPointerEvents.tsx

```diff
@@ -1,21 +1,76 @@
+import * as React from 'react';
 import { useLayoutEffect } from '@radix-ui/react-use-layout-effect';
 
 let changeCount = 0;
 let originalBodyPointerEvents: string;
 
 function useBodyPointerEvents({ disabled }: { disabled: boolean }) {
+  const isTouchOrPenPressedRef = React.useRef(false);
+  const isMousePressedRef = React.useRef(false);
+
+  React.useEffect(() => {
+    const handlePointerDown = (event: PointerEvent) => {
+      const isMouse = event.pointerType === 'mouse';
+      isTouchOrPenPressedRef.current = !isMouse;
+      isMousePressedRef.current = isMouse;
+    };
+    const handlePointerUp = () => {
+      isTouchOrPenPressedRef.current = false;
+      isMousePressedRef.current = false;
+    };
+    document.addEventListener('pointerdown', handlePointerDown);
+    document.addEventListener('pointerup', handlePointerUp);
+    return () => {
+      document.removeEventListener('pointerdown', handlePointerDown);
+      document.removeEventListener('pointerup', handlePointerUp);
+    };
+  }, [isTouchOrPenPressedRef]);
```

Comment:
Hello, I may be wrong about this but I think `useEffect` doesn't run when a ref changes.

On a side note, I've been waiting for this PR. Thanks for doing this!

Reply:
Yes I think you're right, the dependency isn't needed.

Reply:
You are correct, the dependency isn't needed but I always thought we were against empty dependency arrays on this project for some reason. Happy to remove though.

Reply:
I think we just follow the react hook eslint rules, so basically if it moans about something missing, it should be in there. In this case it doesn't moan which means it's safe.